src/query/graphite/native/engine.go

```diff
@@ -32,6 +34,15 @@ type Engine struct {
 
 // NewEngine creates a new query engine.
 func NewEngine(store storage.Storage) *Engine {
+	// Enable pooling.
+	// NB: capacity corresponds to 2 days of data at 10 sec resolution.
+	cap := 2 * 24 * 60 * 6
+	// TODO: take from config.
```

Comment:
Briefly talked about this, I think any meaningful pooling will unfortunately increase the heap size by a substantial amount.

Reply:
Maybe we should #2092 to the comment here? "TODO: take from config #2092"?